subsys/bluetooth/mesh/shell/shell.c

```diff
@@ -1498,3 +1511,45 @@ SHELL_STATIC_SUBCMD_SET_CREATE(mesh_cmds,
 
 SHELL_CMD_ARG_REGISTER(mesh, &mesh_cmds, "Bluetooth mesh shell commands",
 			bt_mesh_shell_mdl_cmds_help, 1, 1);
+
+int bt_mesh_shell_init(void)
+{
+	ctx_shell = shell_backend_uart_get_ptr();
+	printk("Mesh shell initialized\n");
+
+	struct bt_mesh_prov *prov = (struct bt_mesh_prov *)bt_mesh_prov_get();
+
+	/* Return early if internal prov ctx is used. */
+	if (prov == &bt_mesh_shell_prov) {
+		return 0;
+	}
+
+	app_input = prov->input;
```

Comment:
Input action is hardcoded in shell but relaying on app assumes to be any.
The same for output.

Reply:
I'm not sure I understand. Could you elaborate?

Reply:
Take a look at hardcoded oob auth methods: https://github.com/Andrewpini/zephyr/blob/shell_prov_comp/subsys/bluetooth/mesh/shell/shell.c#L374-L378